nexus/test-utils/src/resource_helpers.rs

```diff
@@ -1214,6 +1236,10 @@ impl<'a, N: NexusServer> DiskTest<'a, N> {
     }
 
     /// Returns true if all Crucible resources were cleaned up, false otherwise.
+    ///
+    /// Note: be careful performing this test when also peforming physical disk
```

Comment:
The warning here is that their may be a race with expungement?

Reply:
It's a warning against using this in a test that also performs expungement, no race is required in this case. Basically, doing:

```
<some code that makes disks>

<physical disk expungement>

<region replacement completes>

<delete those disks>

assert!(disk_test.crucible_resources_deleted().await);
```

will always fail (as it will detect the original regions have not been cleaned up) unless the zpool is also manually dropped:

```
<some code that makes disks>

<physical disk expungement>

<region replacement completes>

<delete those disks>

disk_test.remove_zpool(<zpool for expunged physical disk>);

assert!(disk_test.crucible_resources_deleted().await);
```